lib/shopify-cli/process_supervision.rb

```diff
@@ -56,15 +56,23 @@ def for_ident(identifier)
       #
       def start(identifier, args)
         return for_ident(identifier) if running?(identifier)
-        fork do
-          pid_file = new(identifier, pid: Process.pid)
-          pid_file.write
-          STDOUT.reopen(pid_file.log_path, "w")
-          STDERR.reopen(pid_file.log_path, "w")
-          STDIN.reopen("/dev/null", "r")
-          Process.setsid
-          exec(*args)
-        end
+
+        ctx = Context.new
+        pid_file = new(identifier)
+
+        # Make sure the file exists and is empty, otherwise Windows fails
+        File.open(pid_file.log_path, 'w') {}
+        pid = Process.spawn(
+          *args,
+          out: pid_file.log_path,
+          err: pid_file.log_path,
+          in: ctx.windows? ? "nul" : "/dev/null",
```

Comment:
if you are just using the context here maybe we can just do Context.new.windows?